src/anchor.js

```diff
@@ -28,7 +29,34 @@ const aggregators = {
             r += part.r
         }
         return new Point(x / len, y / len, r / len)
-    }
+    },
+    intersect: (config, name, parts) => {
+        const get_line_from_point = (point, offset=1000) => {
+            const p1 = [point.x, point.y]
```

Comment:
To cover the backwards intersection we've talked about, `p1` should probably be something like `[point.x, point.y - offset]`.

Reply:
I think @infused-kim and I understood you preferred to only have a "forward" (i.e. positive on the rotated Y axis) projection of both points to check for intersection. Projecting both on the positive and negative side of the (rotated) Y axis is totally fine and easy to change, but it is also addressed by the anchor ref rotation, like so:

```yaml
points:
    zones:
        point_a:
            key.name: point_a
        point_b:
            key.name: point_b
            anchor:
                - ref: point_a
                  shift: [30,0]
                  rotate: -45
        intersection_point:
            anchor:
                - aggregate:
                  method: intersect
                  parts:
                    - ref: point_a
                      rotate: 180
                    - ref: point_b
                      rotate: 180
```

Up to you to decide on this one. To me, leaving the intersection to be the positive projection only is probably the simplest to reason with.

Reply:
I'd definitely prefer both forward and backward projection. That rotation stuff I mentioned applies to angles other than exactly 180°. And (again, to me) explaining the **lack** of backward projection would be harder. Since two non-parallel, non-colinear lines have exactly one intersection, why should it matter which direction it is? The user selects two [x, y, r°] points, which define two lines, which have one intersection. Would be strange returning no intersection when there would have been one, only the other way around, imho :slightly_smiling_face: